### PR TITLE
Plugin Health Scoring to use GitHub App authentication

### PR DESCRIPTION
**Pull request by alecharp**
- [x] change deployment config for new environment variables
- [x] change values to match environment variables
- [x] create secrets to hold GitHub App private key

The private key of the app must be transformed from `pem` format to `der` format. `openssl` cli tool can be used as described in https://github-api.kohsuke.org/githubappjwtauth.html#Converting_the_private_key_into_a_Java_friendly_format for this.

### CODE REVIEW

**Comment by alecharp** [comment]
Sorry for the forced-push, I had to edit the last commit message. Content of the commits haven't changed.

File: charts/plugin-health-scoring/tests/defaults_test.yaml [hunk 1]
```diff
@@ -67,12 +67,3 @@ tests:
...
         template: deployment.yaml
-      - contains:
-          path: spec.template.spec.containers[0].env
-          content:
-            name: GITHUB_OAUTH
-            valueFrom:
-              secretKeyRef:
-                name: RELEASE-NAME-plugin-health-scoring
-                key: githubOauthToken
-        template: deployment.yaml
```
Comment (dduportal) [inline charts/plugin-health-scoring/tests/defaults_test.yaml:78]:
```suggestion
      - contains:
          path: spec.template.spec.containers[0].env
          content:
            name: GITHUB_APP_PRIVATE_KEY_PATH
            valueFrom:
              secretKeyRef:
                name: RELEASE-NAME-plugin-health-scoring
                key: githubPrivateKey
        template: deployment.yaml
```

File: charts/plugin-health-scoring/tests/custom_values_test.yaml [hunk 2]
```diff
@@ -33,10 +33,6 @@ tests:
           path: data.databasePassword
           value: UGxlYXNlRG9Ob3RVc2VXZWFrUGFzc3dvcmRz # echo 'UGxlYXNlRG9Ob3RVc2VXZWFrUGFzc3dvcmRz' | base64 -d => 'PleaseDoNotUseWeakPasswords'
         template: secret.yaml
-      - equal:
-          path: data.githubOauthToken
-          value: czNjcjN0VG9rZW4= # echo 'czNjcjN0VG9rZW4=' | base64 -d => 's3cr3tToken'
-        template: secret.yaml
```
Comment (dduportal) [inline charts/plugin-health-scoring/tests/custom_values_test.yaml:39]:
```suggestion
      - equal:
          path: data.githubPrivateKey
          value: <to be determined>
        template: secret.yaml
```

File: charts/plugin-health-scoring/templates/secret.yaml [hunk 3]
```diff
@@ -8,4 +8,3 @@ metadata:
 type: Opaque
 data:
   databasePassword: {{ .Values.database.password | b64enc }}
-  githubOauthToken: {{ .Values.config.githubOauth | b64enc }}
```
Comment (dduportal) [inline charts/plugin-health-scoring/templates/secret.yaml:11]:
```suggestion
  githubAppPrivateKey: {{ .Values.config.githubapp.private_key | b64enc }}
```

File: charts/plugin-health-scoring/values.yaml [hunk 4]
```diff
@@ -55,7 +55,10 @@ probes:
 config:
   ucCron: "0 */15 * * * *"
   peCron: "0 10 * * * *"
-  githubOauth: s3cr3tToken
+  github:
+    appId: app-id
+    privateKeyPath: path
```
Comment (dduportal) [inline charts/plugin-health-scoring/values.yaml:60]:
```suggestion
    privateKey: >-
      -------- BEGIN WHATAVER PEM FORMAT -------
      supersecretcontentfopdsvmjfodpbsvùmgkrpbevl
      supersecretcontentfopdsvmjfodpbsvùmgkrpbevl
      supersecretcontentfopdsvmjfodpbsvùmgkrpbevl
      ---------- END WHATEVER PEM FORMAT -------
```

something like this

File: charts/plugin-health-scoring/templates/deployment.yaml [hunk 5]
```diff
@@ -63,11 +63,15 @@ spec:
...
                   key: databasePassword
-            - name: GITHUB_OAUTH
+            - name: GITHUB_APP_ID
+              value: {{ .Values.config.github.appId | squote }}
+            - name: GITHUB_APP_PRIVATE_KEY_PATH
               valueFrom:
                 secretKeyRef:
                   name: {{ include "plugin-health-scoring.fullname" . }}
-                  key: githubOauthToken
+                  key: githubPrivateKey
```
Comment (dduportal) [inline charts/plugin-health-scoring/templates/deployment.yaml:72]: You might want to have a look at https://kubernetes.io/docs/concepts/configuration/secret/#projection-of-secret-keys-to-specific-paths

- The (multiline) key value will be passed through a helm value (see my suggestion to provide a dummy multiline key as default value)
- The value will be base64 encoded (easier to manage multi line strings) and stores in the Kubernetes Secret as a sub key of `data:`
- The deployment will specify the secret (restricted to the sub key) as a volume
- The container of the pod described by the deployment will mount the secret in a predefined path that will be fed to the environment variable here => this "predefined" path can be a helm value (to set in values.yaml) or a static value (to define in the _helpers.tpl file)

File: charts/plugin-health-scoring/tests/defaults_test.yaml [hunk 6]
```diff
@@ -68,11 +68,8 @@ tests:
...
-          content:
-            name: GITHUB_OAUTH
-            valueFrom:
-              secretKeyRef:
-                name: RELEASE-NAME-plugin-health-scoring
-                key: githubOauthToken
+        path: spec.template.spec.containers[0].env
+        content:
+          name: GITHUB_APP_PRIVATE_KEY_PATH
+          value: /mnt/secrets/github/private-key
```
Comment (dduportal) [inline charts/plugin-health-scoring/tests/defaults_test.yaml:74]:
```suggestion
          path: spec.template.spec.containers[0].env
          content:
            name: GITHUB_APP_PRIVATE_KEY_PATH
            value: /mnt/secrets/github/private-key
```

File: charts/plugin-health-scoring/values.yaml [hunk 7]
```diff
@@ -55,10 +55,17 @@ probes:
...
+  github:
+    appId: app-id
+    privateKey: >-
+      -----BEGIN PRIVATE KEY-----
+      blablablabla
+      -----BEGIN PRIVATE KEY-----
+    appInstallationName: jenkinsci
 database:
   username: plugin-health-scoring
   password: s3cr3t
```
Comment (lemeurherve) [inline charts/plugin-health-scoring/values.yaml:67]:
```suggestion
```

**Comment by lemeurherve** [comment]
@dduportal does it looks good to you? If so I'll merge this PR to configure the new GitHub app I've created (cf https://github.com/jenkins-infra/helpdesk/issues/3294)

**Review by lemeurherve — APPROVED** [review]

File: charts/plugin-health-scoring/values.yaml [hunk 8]
```diff
@@ -55,10 +55,17 @@ probes:
...
   ucCron: "0 */15 * * * *"
   peCron: "0 10 * * * *"
-  githubOauth: s3cr3tToken
+  github:
+    appId: app-id
+    privateKey: >-
+      -----BEGIN PRIVATE KEY-----
+      blablablabla
+      -----BEGIN PRIVATE KEY-----
+    appInstallationName: jenkinsci
```
Comment (lemeurherve) [inline charts/plugin-health-scoring/values.yaml:64]: @alecharp Is it used in the Plugins Health Scoring application?

I didn't see any use of it in https://github.com/jenkins-infra/plugin-health-scoring/pull/159/files

I would remove it if it's not currently used.

Comment (alecharp) [reply charts/plugin-health-scoring/values.yaml:64]: It is https://github.com/jenkins-infra/plugin-health-scoring/pull/159/files#diff-7d075ccdbf2d33cad45b93580fbd2097c0730d6e79a2ee5cfc89a0800f48085eR94-R111

Comment (alecharp) [reply charts/plugin-health-scoring/values.yaml:64]: It is here: https://github.com/jenkins-infra/plugin-health-scoring/pull/159/files#diff-7d075ccdbf2d33cad45b93580fbd2097c0730d6e79a2ee5cfc89a0800f48085eR85-R91

Comment (lemeurherve) [reply charts/plugin-health-scoring/values.yaml:64]: Noted, thanks.

Looking at this code and variable name like `privateKeyPEM`, does this mean the key doesn't have to be converted from PEM to DER?

Comment (lemeurherve) [reply charts/plugin-health-scoring/values.yaml:60]: It doesn't seem so, will see.

**Review by lemeurherve — APPROVED** [review]

**Comment by lemeurherve** [comment]
Related: https://github.com/jenkins-infra/helpdesk/issues/3294

**Pull request merged**